sklearn/linear_model/_stochastic_gradient.py

```diff
@@ -2360,8 +2357,8 @@ def partial_fit(self, X, y=None, sample_weight=None):
         self : object
             Returns a fitted instance of self.
         """
+        self._validate_params()
         if not hasattr(self, "coef_"):
```

Comment:
What does this mean? Is it a way of saying "if this estimator hasn't been fitted before"? If yes, then I guess the `self._validate_params()` should remain inside the `if` clause, unlike what I am doing in this PR. Let me know.